src/Sema.zig

```diff
@@ -17686,6 +17691,60 @@ fn reifyStruct(
     return sema.analyzeDeclVal(block, src, new_decl_index);
 }
 
+fn zirAddrSpaceCast(sema: *Sema, block: *Block, extended: Zir.Inst.Extended.InstData) CompileError!Air.Inst.Ref {
```

Comment:
If this builtin is going to exist then `zirPtrCast` should check that the address space is not changed.

Reply:
Good catch, that was the plan. Its implemented